pennylane_lightning/src/bindings/Bindings.cpp

```diff
@@ -324,6 +324,28 @@ void lightning_class_bindings(py::module &m) {
                 const std::vector<size_t> &wires) {
                  return M.expval(operation, wires);
              })
+        .def("generate_samples",
+             [](Measures<PrecisionT> &M, size_t num_shots) {
+                 auto &&result = M.generate_samples(num_shots);
+                 // int num_samples = num_shots;
+                 // std::vector<int> result(num_samples*wires.size(),0);
+
+                 const size_t num_wires = result.size() / num_shots;
```

Comment:
Or simply pass the `num_wires` as an argument in the binding function (defined at line 327) to save the division.